util/detect_bad_urls.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+DISALLOWED_PATTERNS=(
+  "amazonaws\.com"
+  "amazonaws\.com\.cn"
+  "c2s\.ic\.gov"
+  "sc2s\.sgov\.gov"
+)
+
+FILE_TYPES="*.py *.sh"
+
+EXIT_STATUS=0
+
+echo "Scanning for disallowed URL suffixes..."
+
+for FILE_TYPE in $FILE_TYPES; do
+  FILES=$(git ls-files -- $FILE_TYPE)
+
+  for FILE in $FILES; do
+    for PATTERN in "${DISALLOWED_PATTERNS[@]}"; do
+      if grep -E -q "$PATTERN" "$FILE"; then
```

Comment:
With this logic you're scanning the whole repo, but the checker should scan only the diff of the PR.

Reply:
Done